src/editors/EditorContainer.tsx

```diff
@@ -50,7 +60,21 @@ export default function EditorContainer<R, SR>({
   }, [getInputNode]);
 
   // close editor when scrolling
-  useEffect(() => onCommitCancel, [scrollTop, scrollLeft, onCommitCancel]);
+  useEffect(() => {
+    if (scrollLeft !== prevScrollLeft.current || scrollTop !== prevScrollTop.current) {
+      commitCancel();
+    }
+  }, [commitCancel, scrollLeft, scrollTop]);
+
+  // commit changes when editor is closed
+  useEffect(() => {
+    return () => {
+      if (!changeCommitted.current && !changeCanceled.current) {
+        commit();
+      }
+    };
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
Looks dangerous.
I wonder if we do something stupid like this:
```ts
  useEffect(() => () => {
    markAsActuallyUnmounting.current = true;
  }, []);

  useEffect(() => () => {
    if (markAsActuallyUnmounting.current && !changeCommitted.current && !changeCanceled.current) {
      commit();
    }
  });
```

Reply:
I think it is relatively safe to omit `commit` from the dependency list here but I try your suggession